tests/input/semdata.yaml

```diff
@@ -0,0 +1,80 @@
+  semdata: https://he-matchmaker.eu/data/sem/
+  sample: https://he-matchmaker.eu/sample/
+  mat: https://he-matchmaker.eu/material/
+  dm: http://onto-ns.com/meta/characterisation/0.1/SEMImage#
+  parser: http://sintef.no/dlite/parser#
+  gen: http://sintef.no/dlite/generator#
+
+
+# List of documented datasets
+datasets:
+  - "@id": semdata:SEM_cement_batch2/77600-23-001/77600-23-001_5kV_400x_m001
+    "@type": sem:SEMImage
+    title: SEM image of cement
+    description: Back-scattered SEM image of cement sample 77600 from Heidelberg, polished with 1 µm diamond compound.
+    creator: Sigurd Wenner
+    contactPoint: Sigurd Wenner <Sigurd.Wenner@sintef.no>
+    inSeries: semdata:SEM_cement_batch2/77600-23-001
+
+    datamodel: http://onto-ns.com/meta/characterisation/0.1/SEMImage
+    datamodelStorage: https://github.com/HEU-MatCHMaker/DataDocumentation/blob/master/SEM/datamodels/SEMImage.yaml
+    #mappingURL: https://raw.githubusercontent.com/HEU-MatCHMaker/DataDocumentation/refs/heads/master/SEM/datamodels/SEMImage.ttl
+
+    # Contextual documentation of the dataset
+    statements:
+      - ["@id", "sem:fromSample", "sample:SEM_cement_batch2/77600-23-001"]
+      - ["@id", "emmo:isDescriptionOf", "mat:concrete1"]
+
+    distribution:
+      #downloadURL: sftp://nas.aimen.es/P_MATCHMAKER_SHARE_SINTEF/SEM_cement_batch2/77600-23-001/77600-23-001_5kV_400x_m001.tif
+      downloadURL: https://github.com/EMMC-ASBL/tripper/raw/refs/heads/dataset/tests/input/77600-23-001_5kV_400x_m001.tif
+      mediaType: image/tiff
+      parser: parser:sem_hitachi
+      generator: gen:sem_hitachi
+
+  - "@id": semdata:SEM_cement_batch2/77600-23-001
+    "@type": sem:SEMImageSeries
+    title: Series of SEM image of cement sample 77600
+    description: Back-scattered SEM image of cement sample 77600, polished with 1 µm diamond compound.
+    creator: Sigurd Wenner
+    contactPoint: Sigurd Wenner <Sigurd.Wenner@sintef.no>
+    inSeries: semdata:SEM_cement_batch2
+
+    distribution:
+      downloadURL: sftp://nas.aimen.es/P_MATCHMAKER_SHARE_SINTEF/SEM_cement_batch2/77600-23-001
+      mediaType: inode/directory
+
+  - "@id": semdata:SEM_cement_batch2
+    "@type": sem:SEMImageSeries
+    title: Nested series of SEM images of cement batch2
+    description:  ...
+    creator: Sigurd Wenner
+    contactPoint: Sigurd Wenner <Sigurd.Wenner@sintef.no>
+
+    distribution:
+      downloadURL: sftp://nas.aimen.es/P_MATCHMAKER_SHARE_SINTEF/SEM_cement_batch2
+      mediaType: inode/directory
+
+  - "@id": sample:SEM_cement_batch2/77600-23-001
+    "@type": chameo:Sample
+    title: Series for SEM images for sample 77600-23-001.
+
+
+parsers:
+  - "@id": parser:sem_hitachi
+    "@type": oteio:Parser
+    parserType: application/vnd.dlite-parse
+    configuration:
+      driver: hitachi
+
+
+generators:
+  - "@id": gen:sem_hitachi
+    "@type": oteio:Generator
+    generatorType: application/vnd.dlite-generate
+    configuration:
+      driver: hitachi
```

Comment:
What is the generator for an already defined /existing dataset?

Reply:
I think we should not have both parsers and generators for the same thing.

Parsers only for existing data or for data that we know are outputs of specific models.

Generators only for data that does not exist, e.g. inputs to models. 

Parsers and generators in the same file that is supposed to document an existing dataset is very confusing.

Reply:
Good point. I removed the generator from the distribution of the SEM dataset. 

Do you also think it is confusing to keep the declaration of the generator in the YAML file? 
If this file is about documentation of existing datasets, the generator should not be documented in this file. On the other hand, it is easier to manage if corresponding parsers and generators are are documented next to each other.

One thing I did not add in the YAML file was the `sem:SEMImage dcat:distribution value semdata:newdistr` value restriction, where `semdata:newdistr` is the distribution that will be generated.

Reply:
I think Generator and Parsers are typically two very different things. Yes, in our simple examples there is a one to one correpondence. But in most real cases I think that a parser will only capture a subset of all the information available in the dataset (i.e. all the things that are present in the data model, but not all). The generator can only generate a dataset based on what the datamodel describes. 

In this specific case, the generator will never generate a SEM-image, and therefore I think it is confusing to keep in here. 

We have to be very clear about the difference between documenting datasets that have been created by exp or models, and generating new datasets based on (DLite) datamodels. In the first case it is really new, independent data. In the second case it is data that is fetched via a datamodel and repurposed (transformed) to a format (distribution) that we wish for.